### PR TITLE
Dev gpio fix

### PR DESCRIPTION
**Pull request by sjpark608**
related #379 
context manager implementation to close the device file properly

### CODE REVIEW

File: src/edgepi/gpio/edgepi_gpio_chip.py [hunk 1]
```diff
@@ -39,11 +39,10 @@ def read_gpio_pin_state(self, pin_name: str = None):
...
             `bool`: True if state is high, False if state is low
         """
-        self.open_gpio(pin_num=self.__pin_name_dict[pin_name],
+        with self.open_gpio(pin_num=self.__pin_name_dict[pin_name],
                        pin_dir=self.gpiochip_pins_dict[pin_name].dir,
-                       pin_bias=self.gpiochip_pins_dict[pin_name].bias)
-        state = self.read_state()
-        self.close_gpio()
+                       pin_bias=self.gpiochip_pins_dict[pin_name].bias):
+            state = self.read_state()
```
Comment (naomiven) [inline src/edgepi/gpio/edgepi_gpio_chip.py:45]: what does `pin_bias` mean?

File: src/edgepi/gpio/edgepi_gpio_chip.py [hunk 2]
```diff
@@ -83,12 +81,11 @@ def toggle_gpio_pin_state(self, pin_name: str = None):
...
-            self.write_state(False)
-        else:
-            self.write_state(True)
-        self.close_gpio()
+                       pin_bias=self.gpiochip_pins_dict[pin_name].bias):
+            state = self.read_state()
+            if state:
+                self.write_state(False)
+            else:
+                self.write_state(True)
```
Comment (naomiven) [inline src/edgepi/gpio/edgepi_gpio_chip.py:91]: why not have one statement:

```
self.write_state(not state)
```

**Review by naomiven — COMMENTED** [review]
good

File: src/edgepi/gpio/edgepi_gpio_chip.py (hunk 1 above)
Comment (sjpark608) [reply src/edgepi/gpio/edgepi_gpio_chip.py:45]: pin_bias means the pull direction, either pull-down or pull-up

File: src/edgepi/gpio/edgepi_gpio_chip.py (hunk 2 above)
Comment (sjpark608) [reply src/edgepi/gpio/edgepi_gpio_chip.py]: [59fd062](https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/382/commits/59fd0624efdfb778e4b74de00b1df483aca8c43d)

**Review by naomiven — APPROVED** [review]

**Pull request merged**